src/main/java/cz/cvut/kbss/termit/service/business/ExternalVocabularyService.java

```diff
@@ -0,0 +1,232 @@
+                URI uri = new URI(vocabularyIri);
+                Vocabulary vocabulary = null;
+                Optional<Vocabulary> oldVocabulary = repositoryService.find(uri);
+                if (oldVocabulary.isEmpty()) { // new vocabulary  
+                    vocabulary = repositoryService.importVocabulary(uri, RDFFormat.TURTLE.getDefaultMIMEType(), newVocabulary);
+
+                    // add types
+                    vocabulary.addType(cz.cvut.kbss.termit.util.Vocabulary.s_c_pouze_pro_cteni);
+                    vocabulary.addType(cz.cvut.kbss.termit.util.Vocabulary.s_c_externi);
+
+                    final AccessControlList acl = aclService.createFor(vocabulary);
+                    vocabulary.setAcl(acl.getUri());
+
+                    eventPublisher.publishEvent(new VocabularyCreatedEvent(this, vocabulary.getUri()));
+                    LOG.debug("Vocabulary {} import was successful.", vocabularyIri);
+                } else { // updating existing vocabulary
+                    vocabulary = repositoryService.importVocabulary(uri, RDFFormat.TURTLE.getDefaultMIMEType(), newVocabulary);
+                    
+                    // add types
+                    vocabulary.addType(cz.cvut.kbss.termit.util.Vocabulary.s_c_pouze_pro_cteni);
+                    vocabulary.addType(cz.cvut.kbss.termit.util.Vocabulary.s_c_externi);
+
+                    final AccessControlList acl = aclService.createFor(vocabulary);
+                    vocabulary.setAcl(acl.getUri());
+                    
+                    LOG.debug("Vocabulary {} import was successful.", vocabularyIri);
+                }
+                
+                if (firstImportedVocabulary == null) {
+                    firstImportedVocabulary = vocabulary;
+                }
+            }
+        }
+        
+        return firstImportedVocabulary;
+    }
+
+    @Transactional
+    private InputStream downloadExternalVocabulary(String vocabularyIri) throws QueryEvaluationException, RepositoryException{
+
+        List<RdfsResource> response = new ArrayList<>();
+        final Configuration config = context.getBean(Configuration.class);
+
+        String sparqlEndpoint = config.getExternal().getResources();
+        String sparqlQuery = String.format(Utils.loadQuery(EXPORT_FULL_VOCABULARY_QUERY), vocabularyIri);
+
+        SPARQLRepository sparqlRepo = new SPARQLRepository(sparqlEndpoint);
+        sparqlRepo.init();
+        RepositoryConnection conn = sparqlRepo.getConnection();
+        GraphQuery graphQuery = conn.prepareGraphQuery(sparqlQuery);
+
+        GraphQueryResult result = graphQuery.evaluate();
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        Rio.write(result, outputStream, RDFFormat.TURTLE);
+        InputStream vocabularyFile = new ByteArrayInputStream(outputStream.toByteArray());
+        return vocabularyFile;
+
+    }
+    
+    @Scheduled(cron = "${termit.external.reloadCron:-}")
```

Comment:
Missing `@Transactional` causes the reload not to run in a transaction - this causes the added types not to be saved.